freebites/product
# Async Storage & App State #31

# Description

We implemented the functionality for users to save settings, app state, and other necessary data using Async Storage.
We used this functionality to store locally:

- Notification options
- User location
- Post Dietary filters

## Issues

Resolve #31 

## Screenshots

![IMG_1521](https://github.com/freebites/product/assets/143857167/c5590a47-c0f6-483d-b224-bcc3effa9827)
![IMG_1522](https://github.com/freebites/product/assets/143857167/0e60a3b9-f450-446c-9c32-e5225b408b25)

## Test

Test Location:

1. Open app on Expo
2. Go to Home page
3. Click search bar
4. Type in a location of your choice (you should see the console log "updating location:...")
5. Reload the app
6. Click search bar in home page
7. Look at console log and it should display the same location

Test Filters:

1. Open app on Expo
2. Go to Home page
3. Click sidebar icon on the top right
4. Click any filters of your choice (you should see the console log displaying filters chosen)
5. Reload the app
6. Click sidebar icon on the top right
7. Look at console log and it should display the filters you choose before

Test Notifications:

1. Open app on Expo
2. Go to Profile page
3. Go to settings
4. You should see two filters already toggled on which are the first and third
5. Turn on and off as you wish but make sure you change it differently from default
6. Reload the app
7. You should see the changes you made saved
8. You can also go to home page and back and the settings should still persist

## Possible Downsides

Can only store JSON strings.
Added the local storage functionality for sort but sorting functionality is not fully implemented yet.
Only added functionality for notifications in Settings because everything else is not implemented yet.

## Additional Documentations

We used this documentation:
https://react-native-async-storage.github.io/async-storage/docs/api
